src/test/java/ru/marina/tshop/orders/OrderTransactionTest.java

```diff
@@ -0,0 +1,131 @@
+@ContextConfiguration
+public class OrderTransactionTest {
+    @Autowired
+    private ProductDao productDao;
+    @Autowired
+    private OrderDao orderDao;
+    @Autowired
+    private LineItemDao lineItemDao;
+    @Autowired
+    private OrderService orderService;
+    @Autowired
+    private Configuration configuration;
+    @Autowired
+    private IdGenerator idGenerator;
+
+    @org.springframework.context.annotation.Configuration
+    @EnableTransactionManagement
+    @ComponentScan
+    static class Config {
+        @Bean
+        public DataSource dataSource() throws Exception {
+            final EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
+            final EmbeddedDatabase db = builder
+                    .setType(EmbeddedDatabaseType.H2)
+                    .build();
+
+            try (final Connection connection = db.getConnection()) {
+                final Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));
+                final Liquibase liquibase = new Liquibase("test-migration.xml", new ClassLoaderResourceAccessor(), database);
+                liquibase.dropAll();
+                liquibase.update("test");
+            }
+
+            return db;
+        }
+
+        @Bean
+        public PlatformTransactionManager txManager(final DataSource dataSource) {
+            return new DataSourceTransactionManager(dataSource);
+        }
+
+        @Bean
+        public NamedParameterJdbcTemplate namedParameterJdbcTemplate(final DataSource dataSource) {
+            return new NamedParameterJdbcTemplate(dataSource);
+        }
+
+        @Bean
+        public Configuration configSetup() {
+            final Configuration configuration = mock(Configuration.class);
+            when(configuration.getInitialOrderStatus()).thenReturn("created");
+            when(configuration.getNotPaidPaymentStatusId()).thenReturn("psId2");
+            return configuration;
+        }
+
+        @Bean
+        public IdGenerator idGeneratorSetup() {
+            final IdGenerator idGenerator = mock(IdGenerator.class);
+            when(idGenerator.generateId()).thenReturn("mockedId");
+            return idGenerator;
+        }
+
+        @Bean
+        public ProductDao productDao(final NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
+            return new ProductDao(namedParameterJdbcTemplate);
+        }
+    }
+
+    @BeforeEach
+    void setupDB() {
+        productDao.addProduct(new Product("prId1", "tv", "full hd", new BigDecimal("10000.00"), 2, 1, 100, "ctId1"));
+        productDao.addProduct(new Product("prId2", "tv", "full hd", new BigDecimal("12000.00"), 3, 1, 50, "ctId1"));
+        productDao.addProduct(new Product("prId3", "chair", "wooden", new BigDecimal("1000.00"), 1, 1, 20, "ctId2"));
+    }
+
+    @Test
+    public void addOrder() {
+        assertThrows(RuntimeException.class, () -> orderService.addOrder(
```

Comment:
Please see https://github.com/marinakrivonosova/ts-backend/pull/15 for changes in the tests.